furryfeet_be/furryfeet/views.py

```diff
@@ -104,3 +104,156 @@ def post(self, request, pk):
 
         serializer = UserSerializer(user)
         return Response(serializer.data, status=status.HTTP_200_OK)
+    
+
+class DogWalkerView(APIView):
+    def get(self, request):
+        token = request.COOKIES.get('jwt')
+
+        if not token:
+          raise AuthenticationFailed("Unauthenticated!")
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed("Unauthenticated!")
+         
+        try:
+            dog_walker = DogWalker.objects.get(user=request.user)
+            serializer = DogWalkerSerializer(dog_walker)
+            appointments = dog_walker.appointments.all()
+            feedbacks = dog_walker.feedbacks.all()
+            return Response({'dog_walker': serializer.data, 'appointments': AppointmentSerializer(appointments, many=True).data, 'feedbacks': FeedbackSerializer(feedbacks, many=True).data})
+        except DogWalker.DoesNotExist:
+            return Response(status=404)
+
+    def post(self, request, pk):
```

Comment:
the pk here is the path variable, which we don't have in this case and the app fails again. Remomve it from the definition and get the user like we do in the UserView